shared/src/main/scala/com/cibo/evilplot/colors/Coloring.scala

```diff
@@ -188,4 +188,14 @@ object ContinuousColoring {
         )
       }
     }
+  /** Convenience coloring method when we know exactly what the values of the gradients are.
+    * @param colors the colors to use as interpolation points
+    * @param min min value
+    * @param max max override for the data
+    */
+  def gradient(colors: Seq[Color],
```

Comment:
Using the same name but returning a different thing than the other `gradient` methods might be confusing.
E.g. if I'm using a gradient for the `depthColor` PointRenderer:
```scala
val colors: Seq[Color] = Seq(HTMLNamedColors.red, HTMLNamedColors.blue)
val coloring1 = ContinuousColoring.gradient(colors, Some(0.3), Some(0.8), GradientMode.Linear)
val coloring2 = ContinuousColoring.gradient(colors, 0.3, 0.8, GradientMode.Linear)

PointRenderer.depthColor(depths, coloring1) // works
PointRenderer.depthColor(depths, coloring2) // fails
```

Reply:
alright, what other name do you like? We have no name for the type

Reply:
Yeah, it'd be nice if we had something for that already. How about `ColoringFunction`? I've seen that term used elsewhere.
OTOH, what's the use-case for this method? We may need a more formal separation between `Coloring`s and the `ColoringFunction` (or whatever we call them).

Reply:
The use case, as I think that I was describing above, is that we will often want to clip top and bottom of gradients, because plain gradients the way we use them lead to weak results in gaussian point distributions. When you are providing max and min, having to pass a seq that will be thoroughly ignored is terrible API.

I think that Evilplot's approach to coloring and gradients, including which code is public and which code is package private, is very far from ideal, but I did not believe that this PR that provides a convenience method and fixes what I'd consider an honest to goodness bug would not lead to a philosophical discussion. If you'd prefer fewer contributions, I'll be happy to oblige.

Reply:
Your improvements to how gradients handle min & max are absolutely valuable. I don't have any issue with those or the bug fixes. We're always thankful for feedback and PRs and I'm thrilled that people find EvilPlot useful.

My comments were motivated by exactly the problem you mentioned: the public API is not always well defined. Fixing that means paying more attention to API changes than we did in the past. I'm just going off of my general API experience here; resorting to philosophy, in the absence of strong conventions, is a risk.

I see some loose conventions in the current code: accepting `Coloring`s as arguments which are then resolved; and a set of public `gradient` functions that return a `Coloring`. The arguments for existing gradient methods and the new convenient one are also very similar.
Maybe this is a sign that other portions of the color api should be altered or better defined; or that we're not satisfying an important use-case (exposing the coloring function without an intermediate `Coloring`). Either way, it justified some extra attention.

I appreciate that you updated the method name and apologize that this turned into a larger design discussion. I hope it hasn't discouraged you from contributing in the future; we all value your experience in this domain. We're eager to hear feedback about the coloring API (or any part of EP), but please bear with us while conventions are created and the rough edges get polished.